cmd/xl-v1-multipart.go

```diff
@@ -379,6 +379,9 @@ func (xl xlObjects) PutObjectPart(ctx context.Context, bucket, object, uploadID
 	switch size := data.Size(); {
 	case size == 0:
 		buffer = make([]byte, 1) // Allocate atleast a byte to reach EOF
+	case size == -1:
```

Comment:
combine `size == -1 || size > blockSizeV1` here and remove the `default:` clause